linera-chain/src/chain.rs

```diff
@@ -879,9 +924,16 @@ where
             } else {
                 None
             };
+            let refund_grant_to = if grant > Amount::ZERO {
+                raw_outcome.refund_grant_to
+            } else {
+                None
+            };
```

Comment:
(I guess you could use `raw_outcome.refund_grant_to.filter(grant > Amount::ZERO)`. :smile:)

Reply:
fair